src/librustc_mir/transform/qualify_min_const_fn.rs

```diff
@@ -234,31 +238,47 @@ fn check_statement(
 }
 
 fn check_operand(
+    tcx: TyCtxt<'tcx>,
     operand: &Operand<'tcx>,
     span: Span,
+    def_id: DefId,
+    body: &Body<'tcx>
 ) -> McfResult {
     match operand {
         Operand::Move(place) | Operand::Copy(place) => {
-            check_place(place, span)
+            check_place(tcx, place, span, def_id, body)
         }
         Operand::Constant(_) => Ok(()),
     }
 }
 
 fn check_place(
+    tcx: TyCtxt<'tcx>,
     place: &Place<'tcx>,
     span: Span,
+    def_id: DefId,
+    body: &Body<'tcx>
 ) -> McfResult {
     place.iterate(|place_base, place_projection| {
         for proj in place_projection {
             match proj.elem {
                 ProjectionElem::Downcast(..) => {
                     return Err((span, "`match` or `if let` in `const fn` is unstable".into()));
                 }
+                ProjectionElem::Field(..) => {
+                    let base_ty = Place::ty_from(&place.base, &proj.base, body, tcx).ty;
+                    if let Some(def) = base_ty.ty_adt_def() {
```

Comment:
that's less clear than what we have now. I took the rightwards drift on purpose, because the alternative is less readable and the inner block is trivial

Reply:
Alright, fair. Let's agree to disagree re. "readable". :)